hail/python/hailtop/batch/batch.py

```diff
@@ -90,7 +90,8 @@ def __init__(self,
                  default_memory: Optional[str] = None,
                  default_cpu: Optional[str] = None,
                  default_storage: Optional[str] = None,
-                 default_timeout: Optional[Union[float, int]] = None):
+                 default_timeout: Optional[Union[float, int]] = None,
+                 default_shell: str = None):
```

Comment:
I'm pretty sure this won't satisfy mypy it needs to be `Optional[str]`.

Reply:
This is correct.  Note, this behavior changed.  At one point, PEP-848 allowed omitting `Optional` but that is no longer the case.  From https://www.python.org/dev/peps/pep-0484/#union-types:

 > A past version of this PEP allowed type checkers to assume an optional type when the default value is None, as in this code:

> def handle_employee(e: Employee = None): ...

> This would have been treated as equivalent to:

> def handle_employee(e: Optional[Employee] = None) -> None: ...

> This is no longer the recommended behavior. Type checkers should move towards requiring the optional type to be made explicit.

Reply:
Thanks for this Cotton, nice catch Dan!